File: Diagrammatize.Design.Patterns/code/python/Prototype/Sample/UnderlinePen.py
import copy
from framework import Product

class UnderlinePen(Product):
    def __init__(self, ulchar):
        self.ulchar = ulchar

    def use(self, s):
        length = len(s)
        print("\""  + s + "\"")
        print(" ", end="")
        for i in range(length):
            print(self.ulchar, end="")
        print("")

    def createClone(self):
        return copy.deepcopy(self)

File: Diagrammatize.Design.Patterns/code/cxx/Prototype/Sample/UnderlinePen.cpp
#include "UnderlinePen.hpp"
#include <iostream>

UnderlinePen::UnderlinePen(char ulchar_) : ulchar(ulchar_) {}

UnderlinePen::~UnderlinePen() {}

void UnderlinePen::use(const std::string& s) {
    int length = s.length();
    std::cout << "\""  << s << "\"" << '\n';
    std::cout << " ";
    for (int i = 0; i < length; i++) {
        std::cout << ulchar;
    }
    std::cout << '\n';
}

UnderlinePen* UnderlinePen::createClone() {
    return new UnderlinePen(*this);
}
